data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/LogstashMapper.java

```diff
@@ -0,0 +1,42 @@
+package org.opensearch.dataprepper.logstash.mapping;
+
+import com.amazon.dataprepper.model.configuration.PipelineModel;
+import com.amazon.dataprepper.model.configuration.PluginModel;
+import org.opensearch.dataprepper.logstash.exception.LogstashMappingException;
+import org.opensearch.dataprepper.logstash.model.LogstashConfiguration;
+import org.opensearch.dataprepper.logstash.model.LogstashPlugin;
+import org.opensearch.dataprepper.logstash.model.LogstashPluginType;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Converts Logstash configuration model to Data Prepper pipeline model
+ *
+ * @since 1.2
+ */
+public class LogstashMapper {
+    public PipelineModel mapPipeline(LogstashConfiguration logstashConfiguration) {
+        LogstashPluginMapper pluginMapper = new DefaultPluginMapper();
+
+        List<PluginModel> sourcePluginModels = new LinkedList<>();
+        List<PluginModel> prepperPluginModels = new LinkedList<>();
+        List<PluginModel> sinkPluginModels = new LinkedList<>();
+
+        List<LogstashPlugin> inputLogstashPluginList = logstashConfiguration.getPluginSection(LogstashPluginType.INPUT);
```

Comment:
These three lines of code are mostly repeated below.

I'd probably refactor it into a method so that it looks somewhat like this:

```
List<LogstashPlugin> sourcePluginModels = mapPluginSection(logstashConfiguration, LogstashPluginType.INPUT);
if (sourcePluginModels.size() > 1)
             throw new LogstashMappingException("More than 1 source plugins are not supported");

List<LogstashPlugin> prepperPluginModels = mapPluginSection(logstashConfiguration, FILTER);

List<LogstashPlugin> sinkPluginModels = mapPluginSection(logstashConfiguration, OUTPUT);
```